extractor/extractor.cpp

```diff
@@ -262,5 +322,309 @@ int extractor::run()
         SimpleLogger().Write(logWARNING) << e.what();
         return 1;
     }
+
     return 0;
 }
+
+unsigned extractor::CalculateEdgeChecksum(std::unique_ptr<std::vector<EdgeBasedNode>> node_based_edge_list)
+{
+    RangebasedCRC32 crc32;
+    if (crc32.using_hardware())
+    {
+        SimpleLogger().Write() << "using hardware based CRC32 computation";
+    }
+    else
+    {
+        SimpleLogger().Write() << "using software based CRC32 computation";
+    }
+
+    const unsigned crc32_value = crc32(*node_based_edge_list);
+    SimpleLogger().Write() << "CRC32: " << crc32_value;
+
+    return crc32_value;
+}
+
+/**
+    \brief Setups scripting environment (lua-scripting)
+    Also initializes speed profile.
+*/
+void extractor::SetupScriptingEnvironment(lua_State *lua_state, SpeedProfileProperties &speed_profile)
+{
+    // open utility libraries string library;
+    luaL_openlibs(lua_state);
+
+    // adjust lua load path
+    luaAddScriptFolderToLoadPath(lua_state, config.profile_path.string().c_str());
+
+    // Now call our function in a lua script
+    if (0 != luaL_dofile(lua_state, config.profile_path.string().c_str()))
+    {
+        std::stringstream msg;
+        msg << lua_tostring(lua_state, -1) << " occured in scripting block";
+        throw osrm::exception(msg.str());
```

Comment:
All the `stringstreams` here are unnecessary, you can just concat a C-string and a stdlib string:

    { const char* cstring = "abc";  string s;  s += cstring;  cout << s; }

Note: `std::string` has a C-String taking constructor, that invokes undefined behavior if the C-String is null. So if in doubt just default construct it (empty string) and then concat. Otherwise everything works as expected.